public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -104,6 +106,12 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
       headers: {},
     });
 
+    for (const [key, value] of this.customParameters) {
```

Comment:
Can we keep this in the query request handlers? The `_request` is used for label queries too.

Reply:
I've moved this into `performTimeSeriesQuery` and `performInstantQuery`. So data is never null